dandischema/models.py

```diff
@@ -1052,13 +869,13 @@ def check_data(cls, values):
         min_items=1, description="Licenses associated with the item.", nskey="schema"
     )
 
-    citation: TempOptional[str] = Field(readOnly=True, nskey="schema")
+    citation: str = Field(readOnly=True, nskey="schema")
 
     # From assets
-    assetsSummary: TempOptional[AssetsSummary] = Field(readOnly=True, nskey="dandi")
+    assetsSummary: AssetsSummary = Field(readOnly=True, nskey="dandi")
 
     # From server (requested by users even for drafts)
-    manifestLocation: TempOptional[List[HttpUrl]] = Field(readOnly=True, nskey="dandi")
+    manifestLocation: List[HttpUrl] = Field(readOnly=True, nskey="dandi")
```

Comment:
metadata saved does not have to be valid until publish. however, while `assetsSummary` can change and should be updated based on asset related triggers, `manifestLocation` does not change once a dandiset is created and/or published. so it will be considered invalid till those fields are populated. but that's a state a dandiset metadata can be in without much effort (for example no ContactPerson entered or removed).